web/src/app/job/hooks.ts

```diff
@@ -0,0 +1,22 @@
+import { JobTable } from '@/utils/types/job';
+import { useContext, useState } from 'react';
+import { JobListContext } from './context';
+
+export const useJobListContext = (): JobTable => {
+    const jobs = useContext(JobListContext);
+
+    if (jobs === undefined) {
+        throw new Error('Missing JobListContext');
+    }
+
+    return jobs;
+};
```

Comment:
I think we should not export, let's manage context within hooks and pass what we need in the return statement. Also, I can see the context is not used in this hooks so only use it where it is needed.